From 10bb71fb5a4dea41b932431ae4cede58d1be529c Mon Sep 17 00:00:00 2001
From: rmunjewar <rmunjewar@maxlinear.com>
Date: Thu, 17 Oct 2024 11:24:04 +0530
Subject: [PATCH] WLANRTSYS-85288 Increase NL message size

Issue: Driver and hostapd assumed all NL messages sent are under 8K, yet in
legacy afcd development we saw this was not true for the largest afcd messages.
The open source afcd has added code to pass NL messages to driver but it too
was constrained but the default NL message limit being 8K here too.

Fix: Increase the NL message limit to 16K.

FIXUP to WLANRTSYS-63747 Increase default nl80211 msg size

Signed-Off-By : William <washikyan@maxlinear.com>
---
 hostapd/mxl_config.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/hostapd/mxl_config.c b/hostapd/mxl_config.c
index 0130bd24c..9a7b1dd18 100644
--- a/hostapd/mxl_config.c
+++ b/hostapd/mxl_config.c
@@ -2029,7 +2029,12 @@ void mxl_hostapd_config_defaults(struct hostapd_config *conf)
 	 * This is not always enough for the MAX VAPS case, therefore increased
 	 * with possibility to increase it even more with config option.
 	 */
+#ifdef CONFIG_AFC
+	/* AFC NL message to driver can exceed 8192 byte */
+	mxl_conf->nlmsg_default_size = 16 * 1024;
+#else
 	mxl_conf->nlmsg_default_size = 8 * 1024;
+#endif
 
 }
 
-- 
2.43.0

